File: functionapi-content/src/main/java/io/github/cottonmc/functionapi/documentation/ContentCommandDocumentationGenerator.java
package io.github.cottonmc.functionapi.documentation;

import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.tree.RootCommandNode;

import java.io.IOException;
import java.io.PrintWriter;
import java.util.HashMap;
import java.util.Map;

public class ContentCommandDocumentationGenerator {

    public void generate(CommandDispatcher<Map<String, Object>> dispatcher){
        Map<String, Object> contentRegistrationContext = new HashMap<>();
        RootCommandNode<Map<String, Object>> root = dispatcher.getRoot();
        String[] allUsage = dispatcher.getAllUsage(root, contentRegistrationContext, false);

        try {

            PrintWriter fileWriter = new PrintWriter("./functionapi.txt");
            fileWriter.write("Function api registration command reference: "+System.lineSeparator());
            for (String s : allUsage) {
                fileWriter.append(s).append(System.lineSeparator());
            }
            fileWriter.flush();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: functionapi-content/src/main/java/io/github/cottonmc/functionapi/content/documentation/ContentCommandDocumentationGenerator.kt
package io.github.cottonmc.functionapi.content.documentation

import com.mojang.brigadier.CommandDispatcher
import java.io.IOException
import java.io.PrintWriter
import java.io.Writer
import java.util.*

class ContentCommandDocumentationGenerator {

    fun generate(dispatcher: CommandDispatcher<Map<String, Any>>,name:String){
        val fileWriter = PrintWriter("./functionapi/$name.txt")
        generate(dispatcher,fileWriter)
    }

    fun generate(dispatcher: CommandDispatcher<Map<String, Any>>,writer: Writer) {
        val contentRegistrationContext: Map<String, Any> = HashMap()
        val root = dispatcher.root
        val allUsage = dispatcher.getAllUsage(root, contentRegistrationContext, false)
        try {
            writer.write("Function api registration command reference: " + System.lineSeparator())
            for (s in allUsage) {
                writer.append(s).append(System.lineSeparator())
            }
            writer.flush()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}